### ISSUE TITLE
Make formatprg setting optional

### ISSUE THREAD

**Issue by craigp** [elixir-editors/vim-elixir#399]
https://github.com/elixir-editors/vim-elixir/blob/94896fb62cd2e1c004d2e73269c667fb0cb5151a/ftplugin/elixir.vim#L34

This caught me by surprise today, and I don't think this should be forced - it should perhaps be optionally enabled. Someone might not want to use `mix format`; or, like me, might not have moved their current project to 1.6 yet.

**Comment by hauleth** [comment]
This is quasi Optional` as this will not change value if `formatprg` is already set. If you explictly want `formatprg` to be empty then you can opt out by adding `setlocal formatprg=` to `after/ftplugin/elixir.vim` (see `:h after-directory` for more help).

**Comment by craigp** [comment]
I get that, but it's surprising to users, don't you think? They now have to go and set something they never had to set before to keep their editor functioning the way they're used to. In my case I was just trying to format/wrap some doc comments using `gq` like I would normally, and it effectively broke, dumping stdout output into my buffer.

**Comment by hauleth** [comment]
https://xkcd.com/1172/

I believe that things like proper formatter should be set by ftplugin, and if someone want to use non-standard one then it is his/her job to opt-out. `mix format` became default Elixir formatter with 1.6.0 and it should be used by default by vim-elixir.

> They now have to go and set something they never had to set before to keep their editor functioning the way they're used to

Sole fact that you have installed vim-elixir means that you do not want your editor behave the way it used to.

TL;DR

In my opinion it should be **optionally disabled** instead of optionally enabled, and this is exactly the way I did it.

**Comment by craigp** [comment]
Love that comic, and it made your point well 😄 

It's a pity it couldn't check the elixir version you're using before setting it, but I suppose that won't matter soon as things move on. It wasn't just about workflow, it was about error text getting dumped into my editor.

I'll disable `mix format` using formatprg as you suggested.

**Comment by jbodah** [comment]
> It wasn't just about workflow, it was about error text getting dumped into my editor.

I think this is something we should try to tackle. I opened #401 as a follow-up which uses 1.6 but still gets a similar error. I agree that getting that error text dumped into your buffer is not a good experience.

We should also add these steps for opt-out to the README which I can follow up on
